packages/ui/src/components/Table/BaseTable.tsx

```diff
@@ -130,38 +50,47 @@ const BaseTable = <T extends GenericRowStructure>({
         SpinnerComponent = Spinner,
         tableContainerSxProps,
         onRowClick,
+        onRowSelectionChange,
+        onAllRowsSelectionChange,
     } = props;
 
     const {
-        renderCellContent,
-        renderHeadCellContent,
-        makeRowCellId,
-        currentPageRows,
-        filteredRows,
-        handleChangePage,
-        handleRowsPerPageChange,
-        handleSortCellClick,
-        handleChangeSearchValue,
-        handleRowSelection,
-        handleAllRowsSelection,
-        state,
-        selectedRowsCount,
-        headRow,
-        page,
-    } = useTable<T>({
+        state: {
+            currentPageRows,
+            filteredRows,
+            headRow,
+            page,
+            selectedRowsCount,
+            rowsPerPage,
+            searchValue,
+            sortByColumnId,
+            selectedRowIds,
+            sortDirection,
+        },
+        handlers: {
+            handleAllRowsSelection,
+            handleChangePage,
+            handleChangeSearchValue,
+            handleRowSelection,
+            handleRowsPerPageChange,
+            handleSortCellClick,
+            makeRowCellId,
+            renderCellContent,
+            renderHeadCellContent,
+        },
+    } = useTableLogic<T>({
         ...props,
         selectionType,
         onAllRowsSelectionChange,
         onRowSelectionChange,
         defaultRowsPerPage,
     });
 
-    const { searchValue, sortByColumnId, sortDirection, rowsPerPage } = state;
     const hasTableData = Boolean(filteredRows.length);
 
     const { cell: cellClasses } = classes;
 
-    const selectedRowIdsState = props.selectedRowIds ? props.selectedRowIds : state.selectedRowIds;
+    const selectedRowIdsState = selectedRowIdsProp ?? selectedRowIds;
```

Comment:
isn't this `selectedRowIds` origin (internal or external) already decided in useTableLogic ?